broadway_api/handlers/worker.py

```diff
@@ -17,32 +17,43 @@
 class WorkerRegisterHandler(BaseAPIHandler):
     @authenticate_cluster_token
     @schema.validate(
+        on_empty_404=True,
+        input_schema={
+            "type": "object",
+            "properties": {"hostname": {"type": "string"}},
+            "required": ["hostname"],
+            "additionalProperties": False,
+        },
         output_schema={
             "type": "object",
-            "properties": {
-                "worker_id": {"type": "string"},
-                "heartbeat": {"type": "number"},
-            },
-            "required": ["worker_id"],
+            "properties": {"heartbeat": {"type": "number"}},
+            "required": ["heartbeat"],
             "additionalProperties": False,
-        }
+        },
     )
-    def get(self, *args, **kwargs):
+    def post(self, *args, **kwargs):
+        worker_id = kwargs.get("worker_id")
+        hostname = self.body.get("hostname")
+
         worker_node_dao = daos.WorkerNodeDao(self.settings)
+
+        dup = worker_node_dao.find_by_worker_id(worker_id)
+
+        if dup is not None and dup.is_alive:
+            msg = "worker id '{}' already exists".format(worker_id)
+            logger.critical(msg)
```

Comment:
Nit: I do not think we really need to critical log this. We use critical logs for capturing API side errors (ex. when the API enters an invalid state or something unexpected happens). A worker registering with the same ID might not be something that worries us so much. We could reduce this to an `info` log.

Reply:
I see. I'll make it an info